sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -91,7 +91,10 @@ def parse_token(output):
     """
     try:
         token = json.loads(output)
-        dt = datetime.strptime(token["expiresOn"], "%Y-%m-%d %H:%M:%S.%f")
+        expires_on_in_token = token["expiresOn"]
+        if not expires_on_in_token:
+            expires_on_in_token = (datetime.now()+timedelta(days=1)).strftime("%Y-%m-%d %H:%M:%S.%f")
```

Comment:
The token's lifetime depends on the tenant's configuration. We don't know anything about that, so we can't safely guess when the token expires. Underestimating degrades performance; overestimating causes the auth policy to send an expired token. Really this is a CLI bug that should be fixed in the CLI. I think we have two options for handling it here:
1. use the token only once (we can assume it's valid right now)
2. raise AuthenticationFailedError

Option 1 makes the application terribly slow; option 2 maybe crashes it 😄